File: frontend/src/api/quiz.ts
import api from './client';

export interface QuizQuestion {
  id: number;
  english: string;
  options: string[];
  correct_answer: string;
}

export interface QuizAnswer {
  word_id: number;
  selected: string;
}

export interface QuizResult {
  correct: boolean;
  correct_answer: string;
  strength_before: number;
  strength_after: number;
  message: string;
}

export const quizApi = {
  // Получить вопросы
  getQuestions: async (count: number = 10): Promise<{ total: number; questions: QuizQuestion[] }> => {
    const response = await api.get(`/quiz/questions?count=${count}`);
    return response.data;
  },

  // Ответить на вопрос
  answer: async (word_id: number, selected: string): Promise<QuizResult> => {
    const response = await api.post('/quiz/answer', { word_id, selected });
    return response.data;
  },
};

File: backend/app/schemas/quiz.py
from pydantic import BaseModel
from typing import List, Optional


class QuizQuestion(BaseModel):
    """Вопрос для викторины"""
    id: int
    english: str
    options: List[str]  # 4 варианта перевода
    correct_answer: str


class QuizAnswer(BaseModel):
    """Ответ пользователя"""
    word_id: int
    selected: str  # выбранный вариант


class QuizResult(BaseModel):
    """Результат ответа"""
    correct: bool
    correct_answer: str
    strength_before: int
    strength_after: int
    message: str